src/components/CommentSummary/index.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+
+import { gettext } from '../../utils';
+import { Comment, createCommentKey } from '../../reducers/comments';
+import styles from './styles.module.scss';
+
+export type PublicProps = { className?: string; comments: Comment[] };
+
+const CommentSummary = ({ comments, className }: PublicProps) => {
+  const overallComments = [];
+  const fileComments = [];
+
+  for (const comment of comments) {
+    if (comment.filename) {
+      fileComments.push(comment);
+    } else {
+      overallComments.push(comment);
+    }
+  }
+
+  const lines: string[] = [];
+
+  const addComment = (comment: Comment) => {
+    lines.push(comment.content || '');
+    lines.push('');
+  };
+
+  if (overallComments.length) {
+    lines.push(gettext('Overall comment(s):'));
+    lines.push('');
+
+    for (const comment of overallComments) {
+      addComment(comment);
+    }
+  }
+
+  const groups: { [key: string]: Comment[] } = {};
+
+  // Group comments by file / line so they appear under the same header.
+  for (const comment of fileComments) {
+    const key = createCommentKey({
+      fileName: comment.filename,
+      line: comment.lineno,
+    });
+
+    if (!groups[key]) {
+      groups[key] = [];
+    }
+
+    groups[key].push(comment);
+  }
+
+  for (const key of Object.keys(groups)) {
+    let addedHeader = false;
+
+    for (const comment of groups[key]) {
+      if (!addedHeader) {
+        lines.push(gettext('File:'));
+        lines.push(
+          `${comment.filename}${comment.lineno ? `#L${comment.lineno}` : ''}`,
+        );
+        lines.push('');
+        addedHeader = true;
+      }
+      addComment(comment);
+    }
+  }
+
+  const id = 'CommentSummary-textarea';
+
+  return (
+    <form className={className}>
```

Comment:
Just wondering, why did you decide to use a `form` and `textarea` for this, rather than just a `div`? Is it to facilitate copy/paste in the future?

Reply:
No reason. I have an old school habit of always wrapping form controls in `form` but I researched it just now and it's not necessary for anything. All the reasons for still doing it (server side fallbacks, automatic mobile *Go* buttons, serialization, reset behavior...) don't apply here.